ccp-openshift-scan/scanning/rpmverify.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python
+
+# rpm verify utility script for verifying installed RPMs in given env
+# the script has needed functions to to run `rpm -Va` in given environment
+# and filter out config etc files and report the files with issues along
+# with meta like RPM name etc
+
+
+import re
+
+import lib
+
+# Filter the paths you know the resulting image or base image itself
+# has issue about and need to be filtered
+# out since this is a known issue and it is in progress to get fixed.
+FILTER_PATHS = [
+    "/",   # centos base image has issue with following files
+    "/usr/lib/udev/hwdb.d/20-OUI.hwdb",
+    "/usr/lib/udev/hwdb.d/20-acpi-vendor.hwdb",
+    "/usr/lib/udev/hwdb.d/20-bluetooth-vendor-product.hwdb",
+    "/usr/lib/udev/hwdb.d/20-net-ifname.hwdb",
+    "/usr/lib/udev/hwdb.d/20-pci-classes.hwdb",
+    "/usr/lib/udev/hwdb.d/20-pci-vendor-model.hwdb",
+    "/usr/lib/udev/hwdb.d/20-sdio-classes.hwdb",
+    "/usr/lib/udev/hwdb.d/20-sdio-vendor-model.hwdb",
+    "/usr/lib/udev/hwdb.d/20-usb-classes.hwdb",
+    "/usr/lib/udev/hwdb.d/20-usb-vendor-model.hwdb",
+    "/usr/lib/udev/hwdb.d/60-keyboard.hwdb",
+    "/usr/lib/udev/hwdb.d/70-mouse.hwdb",
+    "/usr/lib/udev/hwdb.d/70-touchpad.hwdb",
+    "/usr/lib/udev/hwdb.d/60-evdev.hwdb",
+]
+
+
+# Filter filepaths starting with following directories listing,
+# since these paths are expected to be modified and should not
+# take into account
+
+FILTER_DIRS = [
+    "/var", "/run", "/media", "/mnt", "/tmp", "/proc", "/sys", "/boot"
+]
+
+file_issues_semantics = {
+    "S": "file Size differs",
+    "M": "Mode differs (includes permissions and file type)",
+    "5": "digest (formerly MD5 sum) differs",
+    "D": "Device major/minor number mismatch",
+    "L":  "readLink(2) path mismatch",
+    "U":  "User ownership differs",
+    "G":  "Group ownership differs",
+    "T":  "mTime differs",
+    "P":  "caPabilities differ",
+}
+
+
+class RPMVerify(object):
+    """
+    Verify installed RPMs
+    """
+
+    def get_command(self):
+        """
+        Command to run the rpm verify test
+        """
+        return ["/bin/rpm", "-Va"]
+
+    def get_meta_of_rpm(self, rpm):
+        """
+        Get metadata of given installed package.
+        Metadata captured: SIGPGP, VENDOR, PACKAGER, BUILDHOST
+        """
+        qf = "%{SIGPGP:pgpsig}|%{VENDOR}|%{PACKAGER}|%{BUILDHOST}"
+        cmd = ["/bin/rpm", "-q", "--qf", qf, rpm]
+        out, _ = lib.run_cmd_out_err(cmd)
+        out = out.split("|")
+        return {"RPM": rpm,
+                "SIGNATURE": out[0],
+                "VENDOR": out[1],
+                "PACKAGER": out[2],
+                "BUILDHOST": out[3]
+                }
+
+    def source_rpm_of_file(self, filepath):
+        """
+        Find source RPM of given filepath
+        """
+        cmd = ["/bin/rpm", "-qf", filepath]
+        out, _ = lib.run_cmd_out_err(cmd)
```

Comment:
I don't understand the counter question but I guess you mean that since `rpm -Va` works with RPM db, it won't report issue(s) for a file that's not installed via an rpm, right?

Reply:
@navidshaikh Can you reply?

Reply:
> right?

Correct.